sky/services/oknet/src/org/domokit/oknet/UrlLoaderImpl.java

```diff
@@ -248,9 +249,13 @@ public void onResponse(Response response) {
                     MediaType mediaType = body.contentType();
                     if (mediaType != null) {
                         urlResponse.mimeType = mediaType.type() + "/" + mediaType.subtype();
-                        Charset charset = mediaType.charset();
-                        if (charset != null) {
-                            urlResponse.charset = charset.displayName();
+                        try {
+                          Charset charset = mediaType.charset();
+                          if (charset != null) {
+                              urlResponse.charset = charset.displayName();
+                          }
+                        } catch (UnsupportedCharsetException e) {
+                          Log.e(TAG, "Unsupported charset", e);
```

Comment:
Do we have some way to report these errors back to the dart layer?  For example, right now if you point at a non-existing http endpoint you get some strange MojoInternal error last I tried.